src/autopas/tuning/tuningStrategy/TuningStrategyFactory.cpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file TuningStrategyFactory.cpp
+ * @author seckler
+ * @date 07.02.2020
+ */
+
+#include "TuningStrategyFactory.h"
+
+#include "autopas/options/TuningStrategyOption.h"
+#include "autopas/tuning/tuningStrategy/ActiveHarmony.h"
+#include "autopas/tuning/tuningStrategy/BayesianClusterSearch.h"
+#include "autopas/tuning/tuningStrategy/BayesianSearch.h"
+#include "autopas/tuning/tuningStrategy/MPIParallelizedStrategy.h"
+#include "autopas/tuning/tuningStrategy/PredictiveTuning.h"
+#include "autopas/tuning/tuningStrategy/RandomSearch.h"
+#include "autopas/tuning/tuningStrategy/SlowConfigFilter.h"
+#include "autopas/tuning/tuningStrategy/SortByName.h"
+#include "autopas/tuning/tuningStrategy/TuningStrategyFactoryInfo.h"
+#include "autopas/tuning/tuningStrategy/ruleBasedTuning/RuleBasedTuning.h"
+#include "autopas/tuning/utils/SearchSpaceGenerators.h"
+#include "autopas/utils/NumberSetFinite.h"
+
+namespace autopas::TuningStrategyFactory {
+
+/**
+ * Wraps SearchSpaceGenerators::inferOptionDimensions() and adds a warning about its usage.
+ *
+ * This function acts as a workaround for old and complex tuning strategies that rely on the search space to
+ * be represented as a set of vectors of available options.
+ *
+ * @param searchSpace
+ * @return
+ */
+SearchSpaceGenerators::OptionSpace inferOptionDimensions(const std::set<Configuration> &searchSpace) {
+  AutoPasLog(WARN,
+             "Inferring the dimensions of the search space from the given set of configurations."
+             "This only works reliably if the set was created from a cross product of option vectors.");
+  return SearchSpaceGenerators::inferOptionDimensions(searchSpace);
+}
+
+std::unique_ptr<TuningStrategyInterface> generateTuningStrategy(const std::set<Configuration> &searchSpace,
+                                                                TuningStrategyOption tuningStrategyOption,
+                                                                const TuningStrategyFactoryInfo &info,
+                                                                const std::string &outputSuffix) {
+  std::unique_ptr<TuningStrategyInterface> tuningStrategy = nullptr;
+  switch (static_cast<TuningStrategyOption>(tuningStrategyOption)) {
+    case TuningStrategyOption::randomSearch: {
+      tuningStrategy = std::make_unique<RandomSearch>(info.maxEvidence, std::random_device()());
+      break;
+    }
+
+    case TuningStrategyOption::fullSearch: {
+      utils::ExceptionHandler::exception(
+          "Full Search is no tuning strategy anymore! If you want this behavior don't select any tuning strategy.");
```

Comment:
Theoretically speaking it doesn't have to be an exception. We could even get rid of the `fullSearch` enum value completely. I just wanted AutoPas to be very vocal if one tries to use an old config to avoid "silent" unintentional behavior. A warning is easily overlooked.

Reply:
Okay - I'm fine with this but the example input files in md-flexible still list full search. Potentially this is filtered out by md-flexible and I've not spotted this (but if so, this is a bit confusing). If not, this should be removed from the md-flexible example input files.

Reply:
Ah, I forgot to update all examples. Fixed it and removed it from the options suggested in the doc and auto-completion. 4ee69a4a2